src/message_bus.zig

```diff
@@ -982,7 +982,7 @@ fn MessageBusType(comptime process_type: vsr.ProcessType) type {
                             } else {
                                 // A newer replica connection may even leapfrog this connection and
                                 // then be terminated and set to null before we can get here:
-                                assert(bus.replicas[connection.peer.replica] != null or
+                                assert(bus.replicas[connection.peer.replica] != connection or
```

Comment:
Actually I think this is correct as-is. We do have assertions like this (`assert(x == y or x != y);`) to be explicit that, at this point in the code, we can't narrow down what `x` is.

In this particular case, I think it is saying that in this branch, `bus.replicas[connection.peer.replica]` is either:
- a connection, but not `connection` — i.e., not the same connection we just closed, or
- `null`, because while we were closing this connection, another connection arrived, replaced it, and then closed.

Reply:
Thanks for clarifying that,

I didn't realized that `assert(x == y or x != y)` was a idiomatic thing on TB's codebase.
Knowing that, it makes a lot of sense, and has the same effect as `assert(true)` but it is explicit enough since all previous conditions already have been asserted.

I'm closing this PR.

Reply:
We don't do it often—here's the motivation from our TigerStyle:

> On occasion, you may use a blatantly true assertion instead of a comment as stronger documentation where the assertion condition is critical and surprising.

https://github.com/coilhq/tigerbeetle/blob/main/docs/TIGER_STYLE.md#safety

Reply:
We think of assertions not only as safety checks but also as documentation, especially where a programmer might be tempted to add/remove code that shouldn't be added/removed.